python/pyarrow/ipc.pxi

```diff
@@ -18,6 +18,32 @@
 import warnings
 
 
+cpdef enum MetadataVersion:
+    V1 = <char> CMetadataVersion_V1
+    V2 = <char> CMetadataVersion_V2
+    V3 = <char> CMetadataVersion_V3
+    V4 = <char> CMetadataVersion_V4
+    V5 = <char> CMetadataVersion_V5
+
+
+cdef object _wrap_metadata_version(CMetadataVersion version):
+    return MetadataVersion(<char> version)
```

Comment:
Out of curiosity, where will we break (presumably earlier than this) if we were to encounter an unrecognized version?

Reply:
The fact that integration still passes in the corresponding Java PR #7685 would imply that the version doesn't get checked, no?

Reply:
Sounds like it

Reply:
Well, neither checks, because #7685 passes (Java sending V5 while C++ sends V4) and this passed (Java sends V4 while C++ sends V5)

Though that may be reassuring to anyone planning to use 0.17.1 with 1.0.0.

Reply:
Yikes, well I will open a JIRA about adding appropriate checks at least for 1.0.0

Reply:
See https://github.com/apache/arrow/blob/maint-0.17.x/cpp/src/arrow/ipc/message.cc#L57

So it only checks for old versions but new versions pass silently, which is quite scary to me. But on the other hand the risk of V5 data breaking a V4 application (e.g. Spark) at the moment is low.

Reply:
https://issues.apache.org/jira/browse/ARROW-9399